tests/test_util/aiohttp.py

```diff
@@ -62,11 +65,19 @@ def request(
         if params:
             url = url.with_query(params)
 
-        self._mocks.append(
-            AiohttpClientMockResponse(
-                method, url, status, content, cookies, exc, headers
-            )
+        new_response = AiohttpClientMockResponse(
+            method, url, status, content, cookies, exc, headers, long_poll
         )
+        if long_poll and content_not_empty:
+            for response in self._mocks:
+                if response.long_poll and response.match_request(method, url, params):
```

Comment:
This test still allows me to add responses with:

1. long_poll=False & some content
1. long_poll=True & no-content
1. long_poll=True and some content

and I will always get the content from 1

Reply:
if long_poll=False (default), everything will behave in the way it did. This is very useful for mocking a static request.
for long_poll=True, the first request has to be with an empty content or it will assert. Subsequent ones can have empty content, which will create empty responses.
Also, there is the possibility that i missed something in the code and it doesn't behave like i wanted it to, so please let me know

Reply:
Indeed static requests (long_poll=False) is useful, but just trying to say you should make sure there exist no long_poll=False for this URI when you add long_poll=True. That is why I suggested the assert in the previous code suggestion

Reply:
ok. i understand what you mean now. will fix

Reply:
actually, i am not sure that i fully understand the logic of match_request, but it appears that there could be multiple requests that match (e.g. with wildcards).
the code already assures that it only matches long_poll request objects to future long_poll requests, so not sure i understand what to assert

Reply:
sorry, just figured out where it was in your suggestion.
it can probably be either way. they question is whether someone puts a wildcard request that is static, and would prefer not to assert there if all the long_poll ones go to specific ones.

Reply:
In case of wildcards this seems ok then.